android/src/main/java/com/twiliovoicereactnative/VoiceNotificationReceiver.java

```diff
@@ -0,0 +1,329 @@
+import static com.twiliovoicereactnative.CommonConstants.VoiceEventCallInviteNotificationTapped;
+import static com.twiliovoicereactnative.CommonConstants.VoiceEventCallInviteRejected;
+import static com.twiliovoicereactnative.CommonConstants.VoiceEventType;
+import static com.twiliovoicereactnative.Constants.ACTION_ACCEPT_CALL;
+import static com.twiliovoicereactnative.Constants.ACTION_CALL_DISCONNECT;
+import static com.twiliovoicereactnative.Constants.ACTION_CANCEL_CALL;
+import static com.twiliovoicereactnative.Constants.ACTION_CANCEL_NOTIFICATION;
+import static com.twiliovoicereactnative.Constants.ACTION_FOREGROUND_AND_DEPRIORITIZE_INCOMING_CALL_NOTIFICATION;
+import static com.twiliovoicereactnative.Constants.ACTION_INCOMING_CALL;
+import static com.twiliovoicereactnative.Constants.ACTION_PUSH_APP_TO_FOREGROUND;
+import static com.twiliovoicereactnative.Constants.ACTION_RAISE_OUTGOING_CALL_NOTIFICATION;
+import static com.twiliovoicereactnative.Constants.ACTION_REJECT_CALL;
+import static com.twiliovoicereactnative.Constants.VOICE_CHANNEL_DEFAULT_IMPORTANCE;
+import static com.twiliovoicereactnative.ReactNativeArgumentsSerializer.serializeCall;
+import static com.twiliovoicereactnative.ReactNativeArgumentsSerializer.serializeCallInvite;
+import static com.twiliovoicereactnative.ReactNativeArgumentsSerializer.serializeCancelledCallInvite;
+
+import java.util.Objects;
+
+import android.Manifest;
+import android.app.Notification;
+import android.content.BroadcastReceiver;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.core.app.ActivityCompat;
+import androidx.core.app.NotificationManagerCompat;
+
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.WritableMap;
+import com.twilio.voice.AcceptOptions;
+import com.twilio.voice.Call;
+import com.twilio.voice.CallInvite;
+import com.twilio.voice.CancelledCallInvite;
+
+
+public class VoiceNotificationReceiver extends BroadcastReceiver {
+  @Override
+  public void onReceive(@NonNull Context context, @NonNull Intent intent) {
+    switch (Objects.requireNonNull(intent.getAction())) {
+      case ACTION_INCOMING_CALL:
+        handleIncomingCall(
+          context,
+          intent.getParcelableExtra(Constants.INCOMING_CALL_INVITE),
+          intent.getStringExtra(Constants.UUID));
+        break;
+      case ACTION_ACCEPT_CALL:
+        handleAccept(
+          context,
+          intent.getParcelableExtra(Constants.INCOMING_CALL_INVITE),
+          intent.getStringExtra(Constants.UUID));
+        break;
+      case ACTION_REJECT_CALL:
+        handleReject(
+          context,
+          Objects.requireNonNull(intent.getParcelableExtra(Constants.INCOMING_CALL_INVITE)),
```

Comment:
Should the other cases also use `Objects.requireNonNull`?

Reply:
Done